website_sale_hide_price/models/product_template.py

```diff
@@ -0,0 +1,34 @@
+# Copyright 2020 Tecnativa - Alexandre D. Díaz
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import api, models
+
+
+class ProductTemplate(models.Model):
+    _inherit = "product.template"
+
+    @api.multi
+    def _get_combination_info(
+        self,
+        combination=False,
+        product_id=False,
+        add_qty=1,
+        pricelist=False,
+        parent_combination=False,
+        only_template=False
+    ):
+        """Avoid show prices in other website places.
+        This will show 0.00 value"""
+        ret = super()._get_combination_info(
+            combination=combination,
+            product_id=product_id,
+            add_qty=add_qty,
+            pricelist=pricelist,
+            parent_combination=parent_combination,
+            only_template=only_template)
+        if not self.env.user.partner_id.website_show_price:
+            ret.update(
+                list_price=False,
```

Comment:
Never change the expected type for a variable. This can provoke later problems. At all effects, if a product has 0 price, it's the same as not be set. Change it to 0 both.